cub/test/catch2_test_device_run_length_encode_non_trivial_runs.cu

```diff
@@ -0,0 +1,383 @@
+  REQUIRE(validate_results(in, out_offsets, out_lengths, out_num_runs, num_items));
+}
+
+// Guard against #293
+template <bool TimeSlicing>
+struct device_rle_policy_hub
+{
+  static constexpr int threads = 96;
+  static constexpr int items = 15;
+
+  struct Policy350 : cub::ChainedPolicy<350, Policy350, Policy350>
+  {
+    using RleSweepPolicyT = cub::AgentRlePolicy<threads,
+                                                items,
+                                                cub::BLOCK_LOAD_DIRECT,
+                                                cub::LOAD_DEFAULT,
+                                                TimeSlicing,
+                                                cub::BLOCK_SCAN_WARP_SCANS>;
+  };
+
+  using MaxPolicy = Policy350;
+};
+
+struct CustomDeviceRunLengthEncode {
+  template <bool TimeSlicing,
+            typename InputIteratorT,
+            typename OffsetsOutputIteratorT,
+            typename LengthsOutputIteratorT,
+            typename NumRunsOutputIteratorT>
+  CUB_RUNTIME_FUNCTION __forceinline__ static cudaError_t
+  NonTrivialRuns(void *d_temp_storage,
+                 size_t &temp_storage_bytes,
+                 InputIteratorT d_in,
+                 OffsetsOutputIteratorT d_offsets_out,
+                 LengthsOutputIteratorT d_lengths_out,
+                 NumRunsOutputIteratorT d_num_runs_out,
+                 int num_items,
+                 cudaStream_t stream = 0)
+  {
+    using OffsetT    = int;           // Signed integer type for global offsets
+    using EqualityOp = cub::Equality; // Default == operator
+
+    return cub::DeviceRleDispatch<InputIteratorT,
+                                  OffsetsOutputIteratorT,
+                                  LengthsOutputIteratorT,
+                                  NumRunsOutputIteratorT,
+                                  EqualityOp,
+                                  OffsetT,
+                                  device_rle_policy_hub<TimeSlicing>
+                                 >::Dispatch(d_temp_storage,
+                                             temp_storage_bytes,
+                                             d_in,
+                                             d_offsets_out,
+                                             d_lengths_out,
+                                             d_num_runs_out,
+                                             EqualityOp(),
+                                             num_items,
+                                             stream);
+  }
+};
+
+DECLARE_CDP_WRAPPER(CustomDeviceRunLengthEncode::NonTrivialRuns<true>, run_length_encode_293_true);
+DECLARE_CDP_WRAPPER(CustomDeviceRunLengthEncode::NonTrivialRuns<false>, run_length_encode_293_false);
+
+using bool_types = c2h::type_list<std::true_type,
+                                  std::false_type>;
+
+CUB_TEST("DeviceRunLengthEncode::NonTrivialRuns does not runs out of memory", "[device][run_length_encode]", bool_types)
```

Comment:
suggestion: `bool_types` doesn't look descriptive to me, we might consider `time_slicing`